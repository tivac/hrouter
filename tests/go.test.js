"use strict";

const router = require("../src/router.js");

describe("router", () => {
    beforeEach(() => {
        router._routes = Object.create(null);
    });

    describe(".go()", () => {
        it("should call the callback once the route handlers are complete", () => {
            const fn = jest.fn((ctx, next) => next && next());

            router("/fooga", fn);

            router.go("/fooga", fn);

            expect(fn.mock.calls).toMatchSnapshot();
        });
    });
});
