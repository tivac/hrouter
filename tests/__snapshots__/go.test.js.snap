// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router .go() should call the callback once the route handlers are complete 1`] = `
Array [
  Array [
    Object {
      "url": "/fooga",
    },
    [Function],
  ],
  Array [
    Object {
      "url": "/fooga",
    },
  ],
]
`;
